src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -81,37 +81,447 @@ constexpr const ad_utility::Last<Current, Others...>* unwrapVariant(
 
 // ___________________________________________________________________________
 using PathTuples = ad_utility::sparql_types::PathTuples;
+using VarOrPath = ad_utility::sparql_types::VarOrPath;
+using ObjectList = ad_utility::sparql_types::ObjectList;
+using TripleWithPropertyPath = ad_utility::sparql_types::TripleWithPropertyPath;
+
+vector<TripleWithPropertyPath> SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::TriplesSameSubjectPathContext* ctx) {
+  vector<TripleWithPropertyPath> triples;
+  if (ctx->varOrTerm()) {
+    auto subject = visitTypesafe(ctx->varOrTerm());
+    auto tuples = visitTypesafe(ctx->propertyListPathNotEmpty());
+    for (auto& tuple : tuples) {
+      triples.push_back(
+          TripleWithPropertyPath{subject, tuple.first, tuple.second});
+    }
+  } else if (ctx->triplesNodePath()) {
+    throwCollectionsAndBlankNodePathsNotSupported(ctx->triplesNodePath());
+  }
+  return triples;
+}
+
+// ___________________________________________________________________________
+std::optional<PathTuples> SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::PropertyListPathContext* ctx) {
+  if (ctx->propertyListPathNotEmpty()) {
+    return visitTypesafe(ctx->propertyListPathNotEmpty());
+  } else {
+    return std::nullopt;
+  }
+}
+
+// ___________________________________________________________________________
+PropertyPath SparqlQleverVisitor::visitTypesafe(
+    SparqlAutomaticParser::VerbPathContext* ctx) {
+  PropertyPath p = visitTypesafe(ctx->path());
+  p.computeCanBeNull();
```

Comment:
Can we actually move the `computeCanBeNull` stuff (which can be forgotten) into the PropertyPath constructors?